### ISSUE TITLE
thread-safety related exception in `np.nanpercentile`: TypeError: partition() got an unexpected keyword argument 'axis'

### ISSUE THREAD

**Issue by ogrisel** [colesbury/nogil#59]
I randomly observe the following exceptions when running the following scikit-learn code that use Python-level thread parallelism:

<details>

```python
import sys
from time import perf_counter
import numpy as np
from scipy.sparse import csr_matrix
from sklearn.datasets import fetch_covtype
from sklearn.preprocessing import QuantileTransformer
from sklearn.preprocessing import SplineTransformer
from sklearn.preprocessing import PolynomialFeatures
from sklearn.kernel_approximation import Nystroem
from sklearn.linear_model import SGDClassifier
from sklearn.pipeline import Pipeline
from sklearn.model_selection import RandomizedSearchCV
from sklearn.model_selection import train_test_split
from joblib import parallel_backend
from threadpoolctl import threadpool_limits


nogil = getattr(sys.flags, "nogil", False)
joblib_backend_name = "threading" if nogil else "loky"
n_jobs = 8

print(f"{sys.version=}, {joblib_backend_name=!r}, {n_jobs=}, {nogil=}")

X, y = fetch_covtype(return_X_y=True, as_frame=False)
X = X[:, :5]
X_train, X_test, y_train, y_test = train_test_split(
    X, y, train_size=2_000, test_size=10_000, random_state=0
)

pipeline = Pipeline(
    [
        ("preproc", None),
        ("splines", SplineTransformer()),
        # ("poly", PolynomialFeatures(degree=2, interaction_only=True)),
        ("kernel", Nystroem(n_components=300, gamma=1e-3, random_state=0)),
        (
            "classifier",
            SGDClassifier(
                early_stopping=True,
                validation_fraction=0.2,
                n_iter_no_change=5,
                random_state=0,
            ),
        ),
    ]
)


param_grid = {
    "preproc": [None, QuantileTransformer()],
    "splines__degree": [2, 3, 4],
    "splines__n_knots": [3, 5, 10],
    "kernel__gamma": np.logspace(-5, 3, 100),
    "kernel__n_components": [10, 50, 100, 500],
    "classifier__loss": ["hinge", "log_loss"],
    "classifier__alpha": np.logspace(-5, 1, 100),
}

tic = perf_counter()
with parallel_backend(joblib_backend_name, n_jobs=n_jobs):
    with threadpool_limits(limits=1 if nogil else None):
        search = RandomizedSearchCV(
            pipeline,
            param_distributions=param_grid,
            n_iter=3,
            verbose=1,
            random_state=0,
        ).fit(X_train, y_train)
toc = perf_counter()
print(f"search completed in {toc - tic:.1f}s")
print(search.best_params_)
print(search.score(X_test, y_test))
```

</details>

Here is the full output of a typical failure:

```python-traceback
$ python spline_pipeline_randomized_search.py 
sys.version='3.9.10 (heads/issue56:69f561fa09, May 12 2022, 14:16:43)\n[nogil, GCC 11.2.0]', joblib_backend_name='threading', n_jobs=8, nogil=True
Fitting 5 folds for each of 3 candidates, totalling 15 fits
/home/ogrisel/code/scikit-learn/sklearn/model_selection/_validation.py:378: FitFailedWarning: 
2 fits failed out of a total of 15.
The score on these train-test partitions for these parameters will be set to nan.
If these failures are not expected, you can try to debug them by setting error_score='raise'.

Below are more details about the failures:
--------------------------------------------------------------------------------
1 fits failed with the following error:
Traceback (most recent call last):
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/core/fromnumeric.py", line 57, in _wrapfunc
    return bound(*args, **kwds)
TypeError: take() got an unexpected keyword argument 'axis'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/ogrisel/code/scikit-learn/sklearn/model_selection/_validation.py", line 686, in _fit_and_score
    estimator.fit(X_train, y_train, **fit_params)
  File "/home/ogrisel/code/scikit-learn/sklearn/pipeline.py", line 378, in fit
    Xt = self._fit(X, y, **fit_params_steps)
  File "/home/ogrisel/code/scikit-learn/sklearn/pipeline.py", line 336, in _fit
    X, fitted_transformer = fit_transform_one_cached(
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/joblib/memory.py", line 349, in __call__
    return self.func(*args, **kwargs)
  File "/home/ogrisel/code/nogil/Lib/contextlib.py", line 137, in __exit__
    self.gen.throw(typ, value, traceback)
  File "/home/ogrisel/code/scikit-learn/sklearn/utils/__init__.py", line 933, in _print_elapsed_time
    yield
  File "/home/ogrisel/code/scikit-learn/sklearn/pipeline.py", line 870, in _fit_transform_one
    res = transformer.fit_transform(X, y, **fit_params)
  File "/home/ogrisel/code/scikit-learn/sklearn/base.py", line 870, in fit_transform
    return self.fit(X, y, **fit_params).transform(X)
  File "/home/ogrisel/code/scikit-learn/sklearn/preprocessing/_data.py", line 2597, in fit
    self._dense_fit(X, rng)
  File "/home/ogrisel/code/scikit-learn/sklearn/preprocessing/_data.py", line 2486, in _dense_fit
    self.quantiles_.append(np.nanpercentile(col, references))
  File "<__array_function__ internals>", line 180, in nanpercentile
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/lib/nanfunctions.py", line 1385, in nanpercentile
    return _nanquantile_unchecked(
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/lib/nanfunctions.py", line 1560, in _nanquantile_unchecked
    r, k = function_base._ureduce(a,
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/lib/function_base.py", line 3702, in _ureduce
    r = func(a, **kwargs)
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/lib/nanfunctions.py", line 1582, in _nanquantile_ureduce_func
    result = _nanquantile_1d(part, q, overwrite_input, method)
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/lib/nanfunctions.py", line 1608, in _nanquantile_1d
    return function_base._quantile_unchecked(
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/lib/function_base.py", line 4383, in _quantile_unchecked
    r, k = _ureduce(a,
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/lib/function_base.py", line 3702, in _ureduce
    r = func(a, **kwargs)
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/lib/function_base.py", line 4552, in _quantile_ureduce_func
    result = _quantile(arr,
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/lib/function_base.py", line 4657, in _quantile
    slices_having_nans = np.isnan(
  File "<__array_function__ internals>", line 180, in take
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/core/fromnumeric.py", line 190, in take
    return _wrapfunc(a, 'take', indices, axis=axis, out=out, mode=mode)
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/core/fromnumeric.py", line 66, in _wrapfunc
    return _wrapit(obj, method, *args, **kwds)
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/core/fromnumeric.py", line 43, in _wrapit
    result = getattr(asarray(obj), method)(*args, **kwds)
TypeError: take() got an unexpected keyword argument 'axis'

--------------------------------------------------------------------------------
1 fits failed with the following error:
Traceback (most recent call last):
  File "/home/ogrisel/code/scikit-learn/sklearn/model_selection/_validation.py", line 686, in _fit_and_score
    estimator.fit(X_train, y_train, **fit_params)
  File "/home/ogrisel/code/scikit-learn/sklearn/pipeline.py", line 378, in fit
    Xt = self._fit(X, y, **fit_params_steps)
  File "/home/ogrisel/code/scikit-learn/sklearn/pipeline.py", line 336, in _fit
    X, fitted_transformer = fit_transform_one_cached(
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/joblib/memory.py", line 349, in __call__
    return self.func(*args, **kwargs)
  File "/home/ogrisel/code/nogil/Lib/contextlib.py", line 137, in __exit__
    self.gen.throw(typ, value, traceback)
  File "/home/ogrisel/code/scikit-learn/sklearn/utils/__init__.py", line 933, in _print_elapsed_time
    yield
  File "/home/ogrisel/code/scikit-learn/sklearn/pipeline.py", line 870, in _fit_transform_one
    res = transformer.fit_transform(X, y, **fit_params)
  File "/home/ogrisel/code/scikit-learn/sklearn/base.py", line 870, in fit_transform
    return self.fit(X, y, **fit_params).transform(X)
  File "/home/ogrisel/code/scikit-learn/sklearn/preprocessing/_data.py", line 2597, in fit
    self._dense_fit(X, rng)
  File "/home/ogrisel/code/scikit-learn/sklearn/preprocessing/_data.py", line 2486, in _dense_fit
    self.quantiles_.append(np.nanpercentile(col, references))
  File "<__array_function__ internals>", line 180, in nanpercentile
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/lib/nanfunctions.py", line 1385, in nanpercentile
    return _nanquantile_unchecked(
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/lib/nanfunctions.py", line 1560, in _nanquantile_unchecked
    r, k = function_base._ureduce(a,
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/lib/function_base.py", line 3702, in _ureduce
    r = func(a, **kwargs)
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/lib/nanfunctions.py", line 1582, in _nanquantile_ureduce_func
    result = _nanquantile_1d(part, q, overwrite_input, method)
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/lib/nanfunctions.py", line 1608, in _nanquantile_1d
    return function_base._quantile_unchecked(
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/lib/function_base.py", line 4383, in _quantile_unchecked
    r, k = _ureduce(a,
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/lib/function_base.py", line 3702, in _ureduce
    r = func(a, **kwargs)
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/lib/function_base.py", line 4552, in _quantile_ureduce_func
    result = _quantile(arr,
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/lib/function_base.py", line 4650, in _quantile
    arr.partition(
TypeError: partition() got an unexpected keyword argument 'axis'

  warnings.warn(some_fits_failed_message, FitFailedWarning)
/home/ogrisel/code/scikit-learn/sklearn/model_selection/_search.py:953: UserWarning: One or more of the test scores are non-finite: [   nan 0.5215 0.482 ]
  warnings.warn(
search completed in 0.6s
{'splines__n_knots': 5, 'splines__degree': 4, 'preproc': QuantileTransformer(), 'kernel__n_components': 10, 'kernel__gamma': 0.011768119524349991, 'classifier__loss': 'hinge', 'classifier__alpha': 3.053855508833412e-05}
0.4991
```

Note that the problem is random. Some other times I get a mix of the exception above and another one: `TypeError: take() got an unexpected keyword argument 'out'`:

<details>

```python-traceback
$ python spline_pipeline_randomized_search.py 
sys.version='3.9.10 (heads/issue56:69f561fa09, May 12 2022, 14:16:43)\n[nogil, GCC 11.2.0]', joblib_backend_name='threading', n_jobs=8, nogil=True
Fitting 5 folds for each of 3 candidates, totalling 15 fits
/home/ogrisel/code/scikit-learn/sklearn/model_selection/_validation.py:378: FitFailedWarning: 
7 fits failed out of a total of 15.
The score on these train-test partitions for these parameters will be set to nan.
If these failures are not expected, you can try to debug them by setting error_score='raise'.

Below are more details about the failures:
--------------------------------------------------------------------------------
4 fits failed with the following error:
Traceback (most recent call last):
  File "/home/ogrisel/code/scikit-learn/sklearn/model_selection/_validation.py", line 686, in _fit_and_score
    estimator.fit(X_train, y_train, **fit_params)
  File "/home/ogrisel/code/scikit-learn/sklearn/pipeline.py", line 378, in fit
    Xt = self._fit(X, y, **fit_params_steps)
  File "/home/ogrisel/code/scikit-learn/sklearn/pipeline.py", line 336, in _fit
    X, fitted_transformer = fit_transform_one_cached(
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/joblib/memory.py", line 349, in __call__
    return self.func(*args, **kwargs)
  File "/home/ogrisel/code/nogil/Lib/contextlib.py", line 137, in __exit__
    self.gen.throw(typ, value, traceback)
  File "/home/ogrisel/code/scikit-learn/sklearn/utils/__init__.py", line 933, in _print_elapsed_time
    yield
  File "/home/ogrisel/code/scikit-learn/sklearn/pipeline.py", line 870, in _fit_transform_one
    res = transformer.fit_transform(X, y, **fit_params)
  File "/home/ogrisel/code/scikit-learn/sklearn/base.py", line 870, in fit_transform
    return self.fit(X, y, **fit_params).transform(X)
  File "/home/ogrisel/code/scikit-learn/sklearn/preprocessing/_data.py", line 2597, in fit
    self._dense_fit(X, rng)
  File "/home/ogrisel/code/scikit-learn/sklearn/preprocessing/_data.py", line 2486, in _dense_fit
    self.quantiles_.append(np.nanpercentile(col, references))
  File "<__array_function__ internals>", line 180, in nanpercentile
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/lib/nanfunctions.py", line 1385, in nanpercentile
    return _nanquantile_unchecked(
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/lib/nanfunctions.py", line 1560, in _nanquantile_unchecked
    r, k = function_base._ureduce(a,
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/lib/function_base.py", line 3702, in _ureduce
    r = func(a, **kwargs)
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/lib/nanfunctions.py", line 1582, in _nanquantile_ureduce_func
    result = _nanquantile_1d(part, q, overwrite_input, method)
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/lib/nanfunctions.py", line 1608, in _nanquantile_1d
    return function_base._quantile_unchecked(
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/lib/function_base.py", line 4383, in _quantile_unchecked
    r, k = _ureduce(a,
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/lib/function_base.py", line 3702, in _ureduce
    r = func(a, **kwargs)
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/lib/function_base.py", line 4552, in _quantile_ureduce_func
    result = _quantile(arr,
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/lib/function_base.py", line 4650, in _quantile
    arr.partition(
TypeError: partition() got an unexpected keyword argument 'axis'

--------------------------------------------------------------------------------
2 fits failed with the following error:
Traceback (most recent call last):
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/core/fromnumeric.py", line 57, in _wrapfunc
    return bound(*args, **kwds)
TypeError: take() got an unexpected keyword argument 'axis'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/ogrisel/code/scikit-learn/sklearn/model_selection/_validation.py", line 686, in _fit_and_score
    estimator.fit(X_train, y_train, **fit_params)
  File "/home/ogrisel/code/scikit-learn/sklearn/pipeline.py", line 378, in fit
    Xt = self._fit(X, y, **fit_params_steps)
  File "/home/ogrisel/code/scikit-learn/sklearn/pipeline.py", line 336, in _fit
    X, fitted_transformer = fit_transform_one_cached(
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/joblib/memory.py", line 349, in __call__
    return self.func(*args, **kwargs)
  File "/home/ogrisel/code/nogil/Lib/contextlib.py", line 137, in __exit__
    self.gen.throw(typ, value, traceback)
  File "/home/ogrisel/code/scikit-learn/sklearn/utils/__init__.py", line 933, in _print_elapsed_time
    yield
  File "/home/ogrisel/code/scikit-learn/sklearn/pipeline.py", line 870, in _fit_transform_one
    res = transformer.fit_transform(X, y, **fit_params)
  File "/home/ogrisel/code/scikit-learn/sklearn/base.py", line 870, in fit_transform
    return self.fit(X, y, **fit_params).transform(X)
  File "/home/ogrisel/code/scikit-learn/sklearn/preprocessing/_data.py", line 2597, in fit
    self._dense_fit(X, rng)
  File "/home/ogrisel/code/scikit-learn/sklearn/preprocessing/_data.py", line 2486, in _dense_fit
    self.quantiles_.append(np.nanpercentile(col, references))
  File "<__array_function__ internals>", line 180, in nanpercentile
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/lib/nanfunctions.py", line 1385, in nanpercentile
    return _nanquantile_unchecked(
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/lib/nanfunctions.py", line 1560, in _nanquantile_unchecked
    r, k = function_base._ureduce(a,
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/lib/function_base.py", line 3702, in _ureduce
    r = func(a, **kwargs)
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/lib/nanfunctions.py", line 1582, in _nanquantile_ureduce_func
    result = _nanquantile_1d(part, q, overwrite_input, method)
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/lib/nanfunctions.py", line 1608, in _nanquantile_1d
    return function_base._quantile_unchecked(
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/lib/function_base.py", line 4383, in _quantile_unchecked
    r, k = _ureduce(a,
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/lib/function_base.py", line 3702, in _ureduce
    r = func(a, **kwargs)
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/lib/function_base.py", line 4552, in _quantile_ureduce_func
    result = _quantile(arr,
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/lib/function_base.py", line 4657, in _quantile
    slices_having_nans = np.isnan(
  File "<__array_function__ internals>", line 180, in take
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/core/fromnumeric.py", line 190, in take
    return _wrapfunc(a, 'take', indices, axis=axis, out=out, mode=mode)
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/core/fromnumeric.py", line 66, in _wrapfunc
    return _wrapit(obj, method, *args, **kwds)
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/core/fromnumeric.py", line 43, in _wrapit
    result = getattr(asarray(obj), method)(*args, **kwds)
TypeError: take() got an unexpected keyword argument 'axis'

--------------------------------------------------------------------------------
1 fits failed with the following error:
Traceback (most recent call last):
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/core/fromnumeric.py", line 57, in _wrapfunc
    return bound(*args, **kwds)
TypeError: take() got an unexpected keyword argument 'out'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/ogrisel/code/scikit-learn/sklearn/model_selection/_validation.py", line 686, in _fit_and_score
    estimator.fit(X_train, y_train, **fit_params)
  File "/home/ogrisel/code/scikit-learn/sklearn/pipeline.py", line 378, in fit
    Xt = self._fit(X, y, **fit_params_steps)
  File "/home/ogrisel/code/scikit-learn/sklearn/pipeline.py", line 336, in _fit
    X, fitted_transformer = fit_transform_one_cached(
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/joblib/memory.py", line 349, in __call__
    return self.func(*args, **kwargs)
  File "/home/ogrisel/code/nogil/Lib/contextlib.py", line 137, in __exit__
    self.gen.throw(typ, value, traceback)
  File "/home/ogrisel/code/scikit-learn/sklearn/utils/__init__.py", line 933, in _print_elapsed_time
    yield
  File "/home/ogrisel/code/scikit-learn/sklearn/pipeline.py", line 870, in _fit_transform_one
    res = transformer.fit_transform(X, y, **fit_params)
  File "/home/ogrisel/code/scikit-learn/sklearn/base.py", line 870, in fit_transform
    return self.fit(X, y, **fit_params).transform(X)
  File "/home/ogrisel/code/scikit-learn/sklearn/preprocessing/_data.py", line 2597, in fit
    self._dense_fit(X, rng)
  File "/home/ogrisel/code/scikit-learn/sklearn/preprocessing/_data.py", line 2486, in _dense_fit
    self.quantiles_.append(np.nanpercentile(col, references))
  File "<__array_function__ internals>", line 180, in nanpercentile
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/lib/nanfunctions.py", line 1385, in nanpercentile
    return _nanquantile_unchecked(
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/lib/nanfunctions.py", line 1560, in _nanquantile_unchecked
    r, k = function_base._ureduce(a,
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/lib/function_base.py", line 3702, in _ureduce
    r = func(a, **kwargs)
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/lib/nanfunctions.py", line 1582, in _nanquantile_ureduce_func
    result = _nanquantile_1d(part, q, overwrite_input, method)
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/lib/nanfunctions.py", line 1608, in _nanquantile_1d
    return function_base._quantile_unchecked(
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/lib/function_base.py", line 4383, in _quantile_unchecked
    r, k = _ureduce(a,
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/lib/function_base.py", line 3702, in _ureduce
    r = func(a, **kwargs)
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/lib/function_base.py", line 4552, in _quantile_ureduce_func
    result = _quantile(arr,
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/lib/function_base.py", line 4657, in _quantile
    slices_having_nans = np.isnan(
  File "<__array_function__ internals>", line 180, in take
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/core/fromnumeric.py", line 190, in take
    return _wrapfunc(a, 'take', indices, axis=axis, out=out, mode=mode)
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/core/fromnumeric.py", line 66, in _wrapfunc
    return _wrapit(obj, method, *args, **kwds)
  File "/home/ogrisel/nogil-venv/lib/python3.9/site-packages/numpy/core/fromnumeric.py", line 43, in _wrapit
    result = getattr(asarray(obj), method)(*args, **kwds)
TypeError: take() got an unexpected keyword argument 'out'

  warnings.warn(some_fits_failed_message, FitFailedWarning)
/home/ogrisel/code/scikit-learn/sklearn/model_selection/_search.py:953: UserWarning: One or more of the test scores are non-finite: [  nan   nan 0.482]
  warnings.warn(
search completed in 0.6s
{'splines__n_knots': 10, 'splines__degree': 4, 'preproc': QuantileTransformer(), 'kernel__n_components': 50, 'kernel__gamma': 29.15053062825182, 'classifier__loss': 'log_loss', 'classifier__alpha': 0.1}
0.4847
```

</details>

and other times the program completes without any error. So there must be some kind of race condition.

**Comment by ogrisel** [comment]
In both cases above the exception happens under `np.nanpercentile` so I can try to craft a minimal reproducer if your think this would help.

**Comment by colesbury** [comment]
Fixed in https://github.com/colesbury/numpy/commit/6cbad069f211cc70899a2d73784a8a57ac97a424.

I've uploaded a new patched NumPy 1.22.3 wheel. Since the version number did not change, you'll have to run the following to get the new wheel:

```
pip uninstall numpy
pip cache purge
pip install numpy
```

**Comment by ogrisel** [comment]
I confirm that this is also fixed on my end.
